### PR TITLE
bump 1.2.0

### PR DESCRIPTION
**Pull request by xxchan**

### CODE REVIEW

**Comment by xxchan** [comment]
image

**Comment by BugenZhao** [comment]
https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/

Wow, this is a very recent update.

**Comment by BugenZhao** [comment]
OpenSSL and rdkafka is really a nightmare...

**Comment by xxchan** [comment]
> OpenSSL and rdkafka is really a nightmare...

I think this isn't their fault. More probably apple's. 🤡

**Comment by xxchan** [comment]
Because of this?

image

**Comment by xxchan** [comment]
(x86, failed to build cxx) There's no "-std=c++11" here 🤔️

image

image

**Comment by xxchan** [comment]
> > OpenSSL and rdkafka is really a nightmare...
> 
> I think this isn't their fault. More probably apple's. 🤡

Or homebrew's 🤡

**Comment by xxchan** [comment]
https://github.com/Homebrew/brew/issues/7248 This looks very similar 

> because compiling #include <stdlib.h> produces many warnings.

> PostgreSQL's configure script adds Xcode SDK root -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
Xcode SDK root is printed by xml2-config --cflags, which runs xcrun --show-sdk-path
Homebrew adds CLT SDK root -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk

**Comment by xxchan** [comment]
https://github.com/Homebrew/homebrew-core/pull/51949#issuecomment-601943075

This sounds exactly the root of the problem

> You might have issues if is CMake build is not set up to use `std_cmake_args`.
> 
> CMake is somewhat an unique build tool when it comes to the macOS SDK. In particular, if `CMAKE_OSX_SYSROOT` is not set then CMake makes its own choice of a SDK to use rather than rely on the compiler to handle it (which Homebrew shims). This can generally lead to bad things as Homebrew will still try to use its SDK choice through the compiler shims which means there is now a couple of different SDKs in the include path, and the ordering is likely all wrong too.
>
> If passing `std_cmake_args` is not possible for some reason, then you can try just `-DCMAKE_OSX_SYSROOT`, or perhaps setting the `SDKROOT` env. If the bootstrap doesn’t cope well with either (which would warrant a bug report IMO) then unsetting `HOMEBREW_SDKROOT` should be safe to do here. It wouldn’t be for something like Python which has a python3_config binary.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -40,10 +42,29 @@ jobs:
...
+
+      - name: Post comment 
+        uses: Homebrew/actions/post-comment@master
+        with:
+          token: ${{ github.token }}
+          issue: ${{ github.event.number }}
+          body: "@${{github.actor}} bottle request for ${{ runner.os }} (${{github.event.repository.html_url}}/actions/runs/${{github.run_id}})."
+          bot_body: "Bottle request for ${{ runner.os }} (${{github.event.repository.html_url}}/actions/runs/${{github.run_id}})."
+          bot: BrewTestBot
+
```
Comment (xxchan) [inline .github/workflows/tests.yml:70]: These are a little bit useful, but I still don't know how to let the CI fail

**Comment by BugenZhao** [comment]
What an acute insight you have!

**Comment by xxchan** [comment]
macos-12 failed again... too much headache 🤡 

image

**Comment by BugenZhao** [comment]
Why we get no Monterey 🤔

image

**Comment by xxchan** [comment]
Because macos-12 failed and I’m too lazy to fix it. 🤡

**Comment by BugenZhao** [comment]
> Because macos-12 failed and I’m too lazy to fix it. 🤡

Only Ventura and Sonoma users can take benefits from pre-built bottles and others will suffer from build time of 30 minutes+. I'm okay with that though. 🤣

**Pull request closed without merging**